processor/tailsamplingprocessor/processor.go

```diff
@@ -385,6 +385,7 @@ func (tsp *tailSamplingSpanProcessor) processTraces(resourceSpans ptrace.Resourc
 	for id, spans := range idToSpansAndScope {
 		// If the trace ID is in the sampled cache, short circuit the decision
 		if _, ok := tsp.sampledIDCache.Get(id); ok {
+			tsp.logger.Debug("Trace ID is in the sampled cache", zap.String("id", id.String()))
```

Comment:
wouldn't `zap.Stringer` make sense here? this way, String() isn't called if the Debug level isn't active. Otherwise, we incur in the costs of "String()" for every trace we see.

Reply:
Oh yes! Excellent suggestion, wasn't aware of Stringer.